util/oteltest/src/oteltest/private.py

```diff
@@ -0,0 +1,283 @@
+    sys.path.append(script_path)
+    for script in ls_scripts(script_path):
+        print(f"- Setting up environment for dir {script}")
+        setup_script_environment(temp_dir, script_path, script, wheel_file)
+
+
+def handle_file(script_path, temp_dir, wheel_file):
+    print(f"- Setting up environment for file {script_path}")
+    script_dir = os.path.dirname(script_path)
+    sys.path.append(script_dir)
+    setup_script_environment(
+        temp_dir, script_dir, os.path.basename(script_path), wheel_file
+    )
+
+
+def ls_scripts(script_dir):
+    original_dir = os.getcwd()
+    os.chdir(script_dir)
+    scripts = glob.glob("*.py")
+    os.chdir(original_dir)
+    return scripts
+
+
+def setup_script_environment(venv_parent, script_dir, script, wheel_file):
+    handler = AccumulatingHandler()
+    sink = GrpcSink(handler)
+    sink.start()
+
+    module_name = script[:-3]
+    oteltest_instance: OtelTest = load_test_class_for_script(module_name)()
+
+    script_venv = Venv(str(Path(venv_parent) / module_name))
+    script_venv.create()
+
+    pip_path = script_venv.path_to_executable("pip")
+
+    run_subprocess([pip_path, "install", (wheel_file or "oteltest")])
+
+    for req in oteltest_instance.requirements():
+        print(f"- Will install requirement: '{req}'")
+        run_subprocess([pip_path, "install", req])
+
+    stdout, stderr, returncode = run_python_script(
+        script_dir, script, oteltest_instance, script_venv
+    )
+    print_subprocess_result(stdout, stderr, returncode)
+
+    filename = get_next_json_file(script_dir, module_name)
+    print(f"- Will save telemetry to {filename}")
+    save_telemetry_json(script_dir, filename, handler.telemetry_to_json())
+
+    oteltest_instance.on_stop(handler.telemetry, stdout, stderr, returncode)
+    print(f"- PASSED: {script}")
+
+
+def get_next_json_file(path_str: str, module_name: str):
+    path = Path(path_str)
+    max_index = -1
+    for file in path.glob(f"{module_name}.*.json"):
+        last_part = file.stem.split(".")[-1]
+        if last_part.isdigit():
+            index = int(last_part)
+            if index > max_index:
+                max_index = index
+    return f"{module_name}.{max_index + 1}.json"
+
+
+def save_telemetry_json(script_dir: str, file_name: str, json_str: str):
+    path = Path(script_dir) / file_name
+    with open(str(path), "w", encoding="utf-8") as file:
+        file.write(json_str)
+
+
+def run_python_script(
+    script_dir: str, script: str, oteltest_instance: OtelTest, v
+) -> typing.Tuple[str, str, int]:
+    print(f"- Running python script: {script}")
+    python_script_cmd = [
+        v.path_to_executable("python"),
+        str(Path(script_dir) / script),
+    ]
+
+    wrapper_script = oteltest_instance.wrapper_command()
+    if wrapper_script is not None:
+        python_script_cmd.insert(0, v.path_to_executable(wrapper_script))
+
+    # pylint: disable=R1732
+    proc = subprocess.Popen(
```

Comment:
With oteltest 0.11 I get the following:

```
- Running python script: test_process_runtime_sent_by_default.py
Traceback (most recent call last):
  File "/home/rm/src/test-otel-python/venvtest/bin/oteltest", line 8, in <module>
    sys.exit(main())
  File "/home/rm/src/test-otel-python/venvtest/lib/python3.10/site-packages/oteltest/main.py", line 42, in main
    run(args.script_dir, args.wheel_file, args.venv_parent_dir)
  File "/home/rm/src/test-otel-python/venvtest/lib/python3.10/site-packages/oteltest/private.py", line 37, in run
    setup_script_environment(temp_dir, script_dir, script, wheel_file)
  File "/home/rm/src/test-otel-python/venvtest/lib/python3.10/site-packages/oteltest/private.py", line 67, in setup_script_environment
    stdout, stderr, returncode = run_python_script(
  File "/home/rm/src/test-otel-python/venvtest/lib/python3.10/site-packages/oteltest/private.py", line 112, in run_python_script
    proc = subprocess.Popen(
  File "/usr/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.10/subprocess.py", line 1863, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
PermissionError: [Errno 13] Permission denied: '/tmp/tmpcawnrbcp/test_process_runtime_sent_by_default/bin/'
```

The python_script_cmd is `['/tmp/tmp8frcymoc/test_process_runtime_sent_by_default/bin/', '/tmp/tmp8frcymoc/test_process_runtime_sent_by_default/bin/python', 'tests/integration/test_process_runtime_sent_by_default.py']`

Reply:
Uhm the problem is that in `setup_script_environment` it's not loaded the final instance but the inherited class I have in beetwen my test and OtelTest. The root cause I guess is `load_test_class_for_script` loading the imported class instead of the one defined in the test module :sweat_smile:

Reply:
Worked around by excluding the name of my class explicitly in `is_test_class` but that would probably require an argument in oteltest

Reply:
I was hitting a similar problem (I think) where if the script name (minus ".py") was equal to an available package name, it would fail to load the script. I believe this is fixed now (just pushed here and published to pypi as well).

Reply:
Nope, it's a different issue, my problem is with using an intermediate class that satisfies `is_test_class` checks but it's not a real test case.

Reply:
Ah right. Will see if checking for is not ABC will fix that.

Reply:
it will for an ABC class but I don't have an ABC class.

Reply:
Oh, well I just added a check for `not isabstract`, which I think is good in any case. 😄 

Would it be acceptable to mark your intermediate class as ABC? After this change, that should get it to not be recognized as a test class.

Reply:
Adding an `"OtelTest" in value.__name__` clause works for me